src/cpp/session/include/session/SessionSourceDatabase.hpp

```diff
@@ -188,11 +188,14 @@ core::Error list(std::vector<boost::shared_ptr<SourceDocument> >* pDocs);
 core::Error put(boost::shared_ptr<SourceDocument> pDoc);
 core::Error remove(const std::string& id);
 core::Error removeAll();
+core::Error getPath(const std::string& id, std::string* pPath);
 
 // source database events
 struct Events : boost::noncopyable
 {
    boost::signal<void(boost::shared_ptr<SourceDocument>)> onDocUpdated;
+   boost::signal<void(const std::string&,
```

Comment:
That's a good question ... it is in fact purely additive. Some of the consumers were using these events to keep a cache of id <-> path mappings, so it would have made sense for them to listen to this event too, but since the mapping implementation has since been centralized we're fine there.